src/python_testing/TC_CNET_4_1.py

```diff
@@ -0,0 +1,171 @@
+                      Verify that only one entry has connected status as TRUE \n\
+                      Verify that the number of entries in the Networks attribute is less than or equal to 'MaxNetworksValue'"),
+            TestStep(3, "TH reads ScanMaxTimeSeconds attribute from the DUT",
+                     "Verify that ScanMaxTimeSeconds attribute value is within the range of 1 to 255 seconds"),
+            TestStep(4, "TH reads ConnectMaxTimeSeconds attribute from the DUT",
+                     "Verify that ConnectMaxTimeSeconds attribute value is within the range of 1 to 255 seconds"),
+            TestStep(5, "TH reads InterfaceEnabled attribute from the DUT", "Verify that InterfaceEnabled attribute value is true"),
+            TestStep(6, "TH reads LastNetworkingStatus attribute from the DUT",
+                     "LastNetworkingStatus attribute value will be within any one of the following values \
+                      Success, NetworkNotFound, OutOfRange, RegulatoryError, UnknownError, null"),
+            TestStep(7, "TH reads the LastNetworkID attribute from the DUT. TH reads the Networks attribute from the DUT",
+                     "Verify that LastNetworkID attribute matches the NetworkID value of one of the entries in the Networks attribute list"),
+            TestStep(8, "TH reads the LastConnectErrorValue attribute from the DUT",
+                     "Verify that LastConnectErrorValue attribute value is null"),
+            TestStep(9, "TH reads the SupportedWiFiBands attribute from the DUT",
+                     "Verify that SupportedWiFiBands attribute value has 1 or more entries, all of which are in the range of WiFiBandEnum.")
+        ]
+        return steps
+
+    @run_if_endpoint_matches(has_feature(Clusters.NetworkCommissioning,
+                                         Clusters.NetworkCommissioning.Bitmaps.Feature.kWiFiNetworkInterface))
+    async def test_TC_CNET_4_1(self):
+        # Commissioning already done
+        self.step(1)
+        max_networks_count = await self.read_single_attribute_check_success(
+            cluster=Clusters.NetworkCommissioning,
+            attribute=Clusters.NetworkCommissioning.Attributes.MaxNetworks)
+
+        matter_asserts.assert_int_in_range(max_networks_count, min_value=1, max_value=255, description="MaxNetworks")
+
+        self.step(2)
+        network = await self.read_single_attribute_check_success(cluster=Clusters.NetworkCommissioning, attribute=Clusters.NetworkCommissioning.Attributes.Networks)
+        logger.info(f"network single {network}")
+        networks_dict = await self.read_single_attribute_all_endpoints(
+            cluster=Clusters.NetworkCommissioning,
+            attribute=Clusters.NetworkCommissioning.Attributes.Networks)
+        logger.info("List of network obj")
+        logger.info(list(networks_dict.values()))
+
+        asserts.assert_true(network, "NetworkInfoStruct list should not be empty")
```

Comment:
This will probably need some re-work to accommodate the secondary network device type. Can you check with Andrey and match the format of the CNET-4.3 test?